Partial revert of PR 794

mdinino opened the pull request:
### Description

[MA-2893](https://openedx.atlassian.net/browse/MA-2893)

Partially reverted https://github.com/edx/edx-app-android/pull/794. The sign in and register screens use the alert dialog, but the other screens are using the flying snackbar again.

### Reviewers
If you've been tagged for review, please check your corresponding box once you've given the :+1:.
- [x] Code review: @BenjiLee 
- [x] Code review: @1zaman 
- [ ] Code review: @miankhalid 
- [ ] FYI @andy-armstrong

1zaman commented on OpenEdXMobile/src/main/java/org/edx/mobile/view/LoginActivity.java:
```diff
@@ -257,6 +260,22 @@ public void showNoNetworkDialog() {
+            ViewAnimationUtil.hideMessageBar(activityLoginBinding.errorLayout);
+        }
+        return super.dispatchTouchEvent(ev);
+    }
```
You don't need this implementation if you're not using the fly-in messages in this Activity.

1zaman commented on OpenEdXMobile/src/main/java/org/edx/mobile/view/LoginActivity.java:
```diff
@@ -257,6 +260,22 @@ public void showNoNetworkDialog() {
+    // make sure that on the login activity, all errors show up as a dialog as opposed to a flying snackbar
+    @Override
+    public boolean showErrorMessage(String header, String message) {
+        super.showErrorDialog(header, message);
```
You don't need this since it seems that this method is not being called anywhere in the Activity now.

mdinino replied:
It is public, and I worry that showErrorMessage will be called polymorphicaly. I did it just to be sure that we are showing the dialog every time.

What do you think, should I get rid of it, or leave it just for safety's sake until this is redone?

1zaman replied:
OK, you can leave it like this for now I suppose.

1zaman commented on OpenEdXMobile/src/main/java/org/edx/mobile/view/RegisterActivity.java:
```diff
@@ -376,6 +376,13 @@ public void run() {
+    // make sure that on the login activity, all errors show up as a dialog as opposed to a flying snackbar
+    @Override
+    public boolean showErrorMessage(String header, String message) {
+        super.showErrorDialog(header, message);
```
You don't need this since it seems that this method is not being called anywhere in the Activity now.

mdinino replied:
Same response as above

1zaman commented on OpenEdXMobile/src/main/java/org/edx/mobile/social/SocialLoginDelegate.java:
```diff
@@ -227,7 +225,7 @@ private SocialButtonClickHandler(SocialFactory.SOCIAL_SOURCE_TYPE socialType) {
             if (!NetworkUtil.isConnected(activity)) {
-                callback.showErrorDialog(activity.getString(R.string.no_connectivity),
+                callback.showErrorMessage(activity.getString(R.string.no_connectivity),
                         activity.getString(R.string.network_not_connected));
```
Do you want to show a dialog or fly-in message on social login failure?

mdinino replied:
Dialog. The whole screen is dialog.

1zaman replied:
In that case you should leave the `showErrorDialog()` method here as it is.

mdinino replied:
That's a reference to the callback, which then calls the method in BaseFragmentActivity. I will change it to showAlert() to make it more clear.

mdinino replied:
Actually, here it will definitely be an error, so I'll change it to showError()

1zaman replied:
The callback interface defined only those events that are already implemented in the (base) Activity, so you shouldn't customize it. Just change it back to `showErrorDialog()`, and we should be fine.

BenjiLee commented on OpenEdXMobile/res/layout/activity_login.xml:
```diff
@@ -180,6 +180,16 @@
                     </LinearLayout>
+
+                    <LinearLayout
+                        android:id="@+id/error_layout"
```
I know these are not used, but are they required to be here?

BenjiLee commented:
@mdinino Better commit message. 

`Reverted flying error messages except for Login and Register screens.`

BenjiLee commented:
@mdinino There seem to be a few places where we are putting code that is soon to be deleted. You might want to add a comment in the ticket when we can remove the code. https://openedx.atlassian.net/browse/MA-2412

mdinino commented:
@BenjiLee, @1zaman, @miankhalid I addressed the comments and ready for another pass.

@BenjiLee I'll update https://openedx.atlassian.net/browse/MA-2412 now.

BenjiLee commented:
👍

BenjiLee commented:
@mdinino Better commit message.

Reverted flying error messages except for Login and Register screens.
